test/Sanitizers/sanitizer_coverage.swift

```diff
@@ -0,0 +1,17 @@
+// RUN: %target-build-swift -sanitize=address -sanitize-coverage=edge %s -o %t_binary
+// RUN: rm -rf %t_coverage_dir
+// RUN: mkdir %t_coverage_dir
+// RUN: ASAN_OPTIONS=coverage=1,coverage_dir=%t_coverage_dir %t_binary
+// check the coverage file exists
+// RUN: cat %t_coverage_dir/sanitizer_coverage.swift.tmp_binary.*.sancov > /dev/null
+
+// REQUIRES: executable_test
+// For now restrict this test to platforms where I know this test will pass
```

Comment:
By default, not everyone checks out compiler-rt (even though CI does).
So you would need to make this test conditional on the existence of asan. You can do that by adding:
// REQUIRES: asan_runtime

See swift/test/lit.cfg for more options.